axiom/nr_distributed_trace.c

```diff
@@ -488,7 +488,15 @@ void nr_distributed_trace_set_trace_id(nr_distributed_trace_t* dt,
 
   nr_free(dt->trace_id);
   if (trace_id) {
-    dt->trace_id = nr_strdup(trace_id);
+    int len = nr_strlen(trace_id);
+    if (len < NR_TRACE_ID_SIZE) {
+      int padding = NR_TRACE_ID_SIZE - len;
+      char* dest = (char*)(malloc)(NR_TRACE_ID_SIZE + 1);
+      snprintf(dest, NR_TRACE_ID_SIZE+1, "%0*d%s", padding, 0, trace_id);
+      dt->trace_id = dest;
+    } else {
+      dt->trace_id = nr_strdup(trace_id);
+    }
```

Comment:
What if `trace_id` size is > 32? Is this an error or will OTLP accept such `trace_id`? Maybe it's not part of this PR...

Reply:
trace_id should never be more than 32 characters. Inbound header ids will always be 32 and agent created ids will always be the guid (16) before this PR. If somehow we have bigger trace ids, OTLP compatibility is the least of our worries, as we will create invalid W3C headers.

Reply:
> Inbound header ids will always be 32

This assumption is of course verified somewhere, yes? Again - maybe not part of this PR, but 'inbound' and 'will always be' causes my eyebrow to raise 🤨

Reply:
It turns out that this is not even the code that handles inbound headers. This function is only called in 1 place, with that 1 place being when we are first creating a txn. The only argument ever passed in for the trace id is txn guid, so hypothetically the size comparison isn't even needed (though still should be included).

Reply:
It would be nice to know when how we expect code to be exercised is wrong - though a verbosedebug log seems a tad heavy.

Reply:
The new code does not change the existing behavior - whatever happened with trace_id > 32 will still happen.